server/api/utils/singletons/k8s.py

```diff
@@ -540,6 +541,68 @@ def delete_secrets(
         self.v1api.delete_namespaced_secret(secret_name, namespace)
         return mlrun.common.schemas.SecretEventActions.deleted
 
+    @raise_for_status_code
+    def ensure_configmap(
+        self,
+        resource: str,
+        name: str,
+        data: dict,
+        namespace: str = "",
+        labels: dict = None,
+    ):
+        namespace = self.resolve_namespace(namespace)
+
+        name = f"{resource}-{name}"
```

Comment:
I suggest handling (enforce) the resource name to adhere k8s name length limit (64)
note that if you trim+rand then you would need to make sure you dont create many duplicates every time you "ensure" a too-long-named resource. maybe add a label of "mlrun-function-name" or so, and get existing configmap by that label value to understand whether you need to get or create